modules/jolt_physics/spaces/jolt_contact_listener_3d.cpp

```diff
@@ -193,6 +193,8 @@ bool JoltContactListener3D::_try_add_contacts(const JPH::Body &p_jolt_body1, con
 		return manifolds_by_shape_pair[shape_pair];
 	}();
 
+	ERR_FAIL_COND_V_MSG(!manifold.contacts1.is_empty(), false, "A collision between two sub-shapes resulted in multiple contact manifolds. This should not happen. Please report this.");
```

Comment:
As this is likely going to be difficult to reproduce, it would be nice if some additional information was provided. E.g. which bodies were involved and what their positions/orientations were.

Reply:
Unfortunately I can't include anything particularly useful as far as which bodies were involved, since I can't call `Object::to_string` from a non-main thread. The next best thing is the object's instance ID, aka `ObjectID`, but that will only be relevant in the current session, so would need to be cross-referenced against some prior dump of the physics bodies' instance IDs for it to be useful.

The rest of it is no problem though, if you think it will help.